multicluster/service-mirror/cluster_watcher.go

```diff
@@ -760,6 +878,10 @@ func (rcsw *RemoteClusterServiceWatcher) repairEndpoints(ctx context.Context) er
 	for _, svc := range mirrorServices {
 		updatedService := svc.DeepCopy()
 
+		if svc.Spec.ClusterIP == corev1.ClusterIPNone {
+			rcsw.log.Debugf("Skipped repairing Endpoints for %s/%s", svc.Namespace, svc.Name)
```

Comment:
When we mirror the headless service we have that root headless service and its corresponding endpoints object. This function seems to reconcile gateway addresses and annotations; we would ideally want to skip replacing the subsets for the headless Endpoints (since it will contain the subsets we create based on the available hosts).

Basically, this would still run for all nested clusterIP services (e.g `pod-0-west`) since these point to the gateway -- the headless Endpoints doesn't so we don't want to overwrite its subsets with the gateway address (DNS entries will be misconfigured/inexistent).

I think something that might be worth here is allow the annotation update but if it's headless we perhaps should just skip replacing its subsets. wdyt does that make sense @dadjeibaah? I've added a comment there to make it clearer.

Reply:
Ah ok, that makes sense. Having that explanation helped and including the comment will help in the future as well.